server.go

```diff
@@ -30,8 +32,23 @@ type ServerOptions struct {
 	Authorization     string
 	Placeholder       string
 	PlaceholderImage  []byte
+	Endpoints         Endpoints
 	AlloweOrigins     []*url.URL
-	MaxAllowedSize    int
+}
+
+// Endpoints represents a list of endpoint names to disable.
+type Endpoints []string
+
+// IsValid validates if a given HTTP request endpoint is valid or not.
+func (e Endpoints) IsValid(r *http.Request) bool {
+	parts := strings.Split(r.URL.Path, "/")
+	endpoint := parts[len(parts)-1]
+	for _, name := range e {
```

Comment:
I think this is a fine implementation. However changing this to a map will shave off some ns/op if we really want to go that far.

Alternatively we can change the implementation to use `strings.HasSuffix()`, which is faster than the current `strings.Split()` approach, but not as fast as the `map[string]struct{}` approach.

```go
func (e Endpoints) IsValid(r *http.Request) bool {
	for _, name := range e {
		if strings.HasSuffix(r.URL.Path, name) {
			return false
		}
	}
	return true
}
```

Note that `name` shouldn't be empty otherwise it'll return a false positive.


Backing up the claims: https://gist.github.com/Dynom/7da7f7cefb440f1dce29ba7b377d6f54

Reply:
Wow, thanks for the benchmark. While I think we can change it easily, I also feel we're over-optimizing here a bit. I don't see this as a performance red flag relatively speaking, since for instance, any CGO FFI call implies by default between 50-100ns, without considering memory latency, C side computing...